src/main/java/net/neoforged/neoforge/resource/VanillaClientListeners.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.resource;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import net.minecraft.client.PeriodicNotificationManager;
+import net.minecraft.client.gui.GuiSpriteManager;
+import net.minecraft.client.gui.font.FontManager;
+import net.minecraft.client.particle.ParticleEngine;
+import net.minecraft.client.renderer.CloudRenderer;
+import net.minecraft.client.renderer.GpuWarnlistManager;
+import net.minecraft.client.renderer.LevelRenderer;
+import net.minecraft.client.renderer.ShaderManager;
+import net.minecraft.client.renderer.block.BlockRenderDispatcher;
+import net.minecraft.client.renderer.blockentity.BlockEntityRenderDispatcher;
+import net.minecraft.client.renderer.entity.EntityRenderDispatcher;
+import net.minecraft.client.renderer.texture.TextureManager;
+import net.minecraft.client.resources.FoliageColorReloadListener;
+import net.minecraft.client.resources.GrassColorReloadListener;
+import net.minecraft.client.resources.MapDecorationTextureManager;
+import net.minecraft.client.resources.MobEffectTextureManager;
+import net.minecraft.client.resources.PaintingTextureManager;
+import net.minecraft.client.resources.SplashManager;
+import net.minecraft.client.resources.language.LanguageManager;
+import net.minecraft.client.resources.model.EquipmentAssetManager;
+import net.minecraft.client.resources.model.ModelManager;
+import net.minecraft.client.sounds.SoundManager;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.PreparableReloadListener;
+import net.neoforged.neoforge.client.event.AddClientReloadListenerEvent;
+import net.neoforged.neoforge.common.util.VanillaClassToKey;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Keys for vanilla {@link PreparableReloadListener reload listeners}, used to specify dependency ordering in the {@link AddClientReloadListenerEvent}.
+ * <p>
+ * Due to the volume of vanilla listeners, these keys are automatically generated based on the class name.
+ * 
+ * @see {@link VanillaServerListeners} for vanilla server listener names.
+ * @see {@link NeoForgeReloadListeners} for Neo-added listener names.
+ */
+public class VanillaClientListeners {
```

Comment:
Should be moved to a package under `net.neoforged.neoforge.client`.

Reply:
Guessing that needs to be moved for #1920 

Sucks for discoverability, but not much else to do about it. Links from the other classes will suffice.

Reply:
Indeed. And the links in the other classes will break with split sources but that's no big deal.